src/Aspect/ContractCheckerAspect.php

```diff
@@ -32,6 +32,11 @@ class ContractCheckerAspect implements Aspect
     private $reader = null;
 
     /**
+     * @var MethodInvocation
+     */
+    private $invocation;
```

Comment:
I think that private property here can be incorrect for recursive calls of methods, etc, this adds some extra overhead for assigning the value to the property for each method call, maybe it will be better to rewrite code as following:
 - rename `isContractSatisfied` to `ensureContractSatisfied` that will throw simple `DomainException` if check was failed (bool false)
 - change all ifs in advice's  body to `try...catch` construction that rethrowing of `ContractViolation` exception
 - all exceptions now will be in one place, no need for property and extra argument